test/steepfile_test.rb

```diff
@@ -183,6 +183,39 @@ def test_collection_explicit
     end
   end
 
+  def test_collection_missing
+    in_tmpdir do
+      current_dir.join('rbs_collection.yaml').write('')
+      project = Project.new(steepfile_path: current_dir + "Steepfile")
+
+      assert_output nil, /rbs collection/ do
+        Project::DSL.parse(project, <<~EOF)
+          target :app do
+          end
+        EOF
+      end
+
+      assert_nil project.targets[0].options.collection_lock
+    end
+  end
+
+  def test_collection_bad
+    in_tmpdir do
+      current_dir.join('rbs_collection.yaml').write('')
+      current_dir.join('rbs_collection.lock.yaml').write(']')
+      project = Project.new(steepfile_path: current_dir + "Steepfile")
+
+      assert_output nil, /syntax/i do
+        Project::DSL.parse(project, <<~EOF)
+          target :app do
+          end
+        EOF
+      end
+
+      assert_nil project.targets[0].options.collection_lock
+    end
+  end
+
```

Comment:
There’s a good chunk of WET in this file. Other questionable choices:
* Giving up indents rather than utilizing `<<~`
* Using `#tap` instead of `#then` despite the `void` return

My tests also check the IO output since – how else do I `assert` a warning `¯\_(ツ)_/¯`